search.py

```diff
@@ -456,12 +471,77 @@ def __call__(self, percept):
         self.s = current_state
         return self.a
 
+# ______________________________________________________________________________
+
+class OneDimStateSpaceProblem(Problem):
+    """ Fig. [4.23]
+    """
+    def __init__(self):
+        raise NotImplementedError
+
+    def h(self, state):
+        """
+        returns least possible cost for the given state
+        """
+        raise NotImplementedError
+
+    def c(self, s, a, s1):
+        """
+        returns a cost estimate to move from state 's' to state 's1'
+        """
+        raise NotImplementedError
+
+
     def update_state(self, percept):
         raise NotImplementedError
 
-def lrta_star_agent(s1):
-    "[Fig. 4.24]"
-    unimplemented()
+
+class LRTAStarAgent:
+
+    """Fig. [4.24]
+    Abstract class for LRTA*-Agent. A problem needs to be
+    provided which is an instanace of a subclass of Problem Class.
+
+    Takes a OneDimStateSpaceProblem Fig. [4.23] as a problem
+    """
+
+    def __init__(self, problem):
+        self.problem = problem
+        self.result = {}
+        self.H = {}
+        self.s = None
+        self.a = None
+
+    def __call__(self, s1):
+        current_state = self.update_state(s1)
```

Comment:
@SnShine I guess there is a slight error here. In your implementation there is no update_state method present but it is being called.